zkevm-circuits/src/evm_circuit/execution/create.rs

```diff
@@ -177,23 +185,40 @@ impl<F: Field> ExecutionGadget<F> for CreateGadget<F> {
             );
         });
 
-        let transfer = TransferGadget::construct(
-            cb,
-            from_bytes::expr(&caller_address.cells),
+        // TODO：check for address collision case by code hash previous
+        cb.account_read(
             new_address.clone(),
-            0.expr(),
-            1.expr(),
-            value.clone(),
-            &mut callee_reversion_info,
+            AccountFieldTag::CodeHash,
+            code_hash_previous.expr(),
         );
 
-        cb.account_write(
-            new_address.clone(),
-            AccountFieldTag::Nonce,
-            1.expr(),
-            0.expr(),
-            Some(&mut callee_reversion_info),
+        let not_address_collision = IsZeroGadget::construct(cb, code_hash_previous.expr());
+        cb.require_zero(
```

Comment:
If IsZeroGadget is correct，we don't need this check？

Reply:
update to check op code is create2 when address collision case. [cb6e590](https://github.com/scroll-tech/zkevm-circuits/pull/406/commits/cb6e590d00ab8b8c8d1416ac526cb29b2d6d59e7) thanks!